src/cocoa/toga_cocoa/widgets/canvas.py

```diff
@@ -152,6 +152,10 @@ def stroke(self, color, line_width, draw_context, *args, **kwargs):
         else:
             # Set color to black
             core_graphics.CGContextSetRGBStrokeColor(draw_context, 0, 0, 0, 1)
+        if line_dash is not None:
+            core_graphics.CGContextSetLineDash(draw_context, 0, (CGFloat*len(line_dash))(*line_dash), 2)
```

Comment:
@bryall Can you help me understand this line. I understand that the definition of CGContextSetLineDash is:
  
    void CGContextSetLineDash(CGContextRef c, CGFloat phase, const CGFloat *lengths, size_t count);

I understand that *line_dash unpacks the list, why did you then multiply it by the length of the list as a CGFloat?

I'm still a beginner at using the Cocoa APIs, and it isn't quite making sense to me. Thanks!

Reply:
I got the solution from stack overflow (still learning cocoa too). it is creating a c_double array object and initializing it with the pass in array. Thanks for making a comment, totally missed the hard coded length of array for the last argument that I just fixed